books/RayTracingTheNextWeek.html

```diff
@@ -2232,7 +2232,7 @@
 
         color value(double u, double v, const point3& p) const override {
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++ highlight
-            return color(1,1,1) * 0.5 * (1 + sin(scale*p.z() + 10*noise.turb(p)));
+            return color(1,1,1) * 0.5 * (1 + sin(scale*p.z() + 10*noise.turb(scale*p)));
```

Comment:
Rather than scaling the Z component twice, let's create a vector to hold the scaled result, and then use that:
```
auto s = scale * p;
return color(1,1,1) * 0.5 * (1 + sin(s.z() + 10*noise.turb(s)));
```

This analogous change (defining a temporary scaled point) should be made in the earlier listing `noise-tex-2`.

Reply:
Done